### ISSUE TITLE
Module version mismatch. Expected 49, got 48

### ISSUE THREAD

**Issue by ingsaurabh** [agnat/node_mdns#173]
Using electron npm start I get this

```
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Module version mismatch. Expected 49, got 48.
    at Error (native)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:167:20)
    at Object.Module._extensions..node (module.js:568:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:167:20)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/ingsaurabh/dp/node_modules/mdns/lib/dns_sd.js:24:20)
```

more info 
i```
ngsaurabh@ingsaurabh:~/dp$ electron --version
v1.3.3
ingsaurabh@ingsaurabh:~/dp$ node --version
v4.2.6
ingsaurabh@ingsaurabh:~/dp$ nodejs --version
v4.2.6
```

**Comment by ronkorving** [comment]
That means you're running node-mdns with a version of Node ABI incompatible with the one you built the dependency with. Make sure to not change Node (major) versions between running the app and building the dependencies.

**Comment by ingsaurabh** [comment]
Can you please tell me how to fix it in layman terms as I am new to nodejs and electron

**Comment by ingsaurabh** [comment]
Ok fix the above problem by 
electron-rebuild command in node_modules/./bin folder,  now too many warnings

`*** WARNING *** The program 'electron' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=electron>
*** WARNING *** The program 'electron' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=electron&f=DNSServiceRegister>
*** WARNING *** The program 'electron' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=electron>
*** WARNING *** The program 'electron' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=electron&f=DNSServiceRegister>

`

**Comment by agnat** [comment]
The warnings are from the compatibility layer. Please follow the link for instructions on how to disable (some of) them.
